app/javascript/controllers/show_more_button_controller.js

```diff
@@ -0,0 +1,43 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="show-more-button"
+export default class extends Controller {
+  connect() {
+    this.element.textContent = 'Show more'
+    this.element.classList.add('btn', 'btn-link', 'p-0', 'mb-3', 'show-more-button')
+    this.element.ariaDisabled = true
+    this.element.ariaExpaned = false
+    this.element.ariaLabel = "This button is disabled because assistive technologies already announced the content."
+    this.element.prepend(this.buttonIcon())
+  }
+
+  isTruncated() {
+    return !this.element.ariaExpaned
+  }
+
+  buttonIcon() {
+    const icon = document.createElement('i')
+    icon.ariaHidden = true
+    icon.className = 'bi me-1'
+
+    if (this.isTruncated()) {
+      icon.classList.add('bi-arrow-down-circle-fill')
+    } else {
+      icon.classList.add('bi-arrow-up-circle-fill')
+    }
+
+    return icon
+  }
+
+  toggle() {
+    if (this.isTruncated()) {
+      this.element.innerText = "Show less"
+      this.element.ariaExpaned = true
+    } else {
+      this.element.innerText = "Show more"
+      this.element.ariaExpaned = false
+    }
+
+    this.element.prepend(this.buttonIcon())
```

Comment:
Does this cause a new icon node to be created every time the toggle method is run?

Reply:
This is the old code but i assume no because we are resting the innerHTML in lines 34 and 37.

Reply:
Ah, thanks!